modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/provider/ObjectEntryInfoItemFieldValuesProvider.java

```diff
@@ -248,6 +254,15 @@ else if (Validator.isNotNull(objectField.getRelationshipType())) {
 				return objectEntry.getTitleValue();
 			}
 		}
+		else if (Objects.equals(objectField.getType(), "Date")) {
```

Comment:
You will also need to include this logic in class com.liferay.object.web.internal.info.item.renderer.ObjectEntryRowInfoItemRenderer#_getValues, as it is used to display the Object information when a table template is configured in List Style property on Collection Display fragment.